server/gui/workers.py

```diff
@@ -1,47 +1,94 @@
+from multiprocessing import Process
 import traceback
+import time
 
-from server.gui.utils import WorkerSignals
+import requests
 
+from server.gui.utils import SiteReadySignals
 
-class DataLoadWorker():
-    def __init__(self, data_file, layout="umap", *args, **kwargs):
-        super(DataLoadWorker, self).__init__()
+
+class EmittingProcess(Process):
+    def __init__(self, parent_conn, child_conn, *arg, **kwargs):
+        super(EmittingProcess, self).__init__()
+        self.parent_conn = parent_conn
+        self.child_conn = child_conn
+
+    def run(self):
+        self.parent_conn.close()
+
+    def emit(self, signal_name, *args):
+        message = (signal_name, *args)
+        self.child_conn.send(message)
+
+
+class Worker(EmittingProcess):
+    def __init__(self, parent_conn, child_conn, data_file, title, host, port, layout=["umap"], *args, **kwargs):
+        super(Worker, self).__init__(parent_conn, child_conn)
         self.data_file = data_file
         self.layout = layout
-        self.signals = WorkerSignals()
+        self.title = title
+        self.host = host
+        self.port = port
 
     def run(self):
+        super(Worker, self).run()
         if not self.data_file:
-            self.signals.finished.emit()
+            self.emit("finished")
```

Comment:
strictly a style issue:  I remember from a previous PR review that you wanted all strings like this to be in constant variables, eg, `SIG_FINISHED`.    Should these be variables, or inline strings (I personally *like* inline strings where the variable name adds no additional value).

Reply:
yeah, after trying things "Colin style" as I like to call it, I like using strings better.